### PR TITLE
[Components with more than 1 service] LPS-195086 Refactor DLSizeLimitManagedServiceFactory component

### PR DESCRIPTION
**Pull request by RenanUmV**

### CODE REVIEW

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:sf

**Comment by RenanUmV** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java [hunk 1]
```diff
@@ -6,144 +6,38 @@
...
  * @author Adolfo Pérez
  */
-@Component(
-	configurationPid = "com.liferay.document.library.internal.configuration.DLSizeLimitConfiguration",
-	property = Constants.SERVICE_PID + "=com.liferay.document.library.internal.configuration.DLSizeLimitConfiguration.scoped",
-	service = {
-		DLSizeLimitManagedServiceFactory.class, ManagedServiceFactory.class
-	}
-)
+@Component(service = ManagedServiceFactory.class)
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java:27]: you forgot the servicePid property

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/helper/DLSizeLimitConfigurationHelper.java [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Renan Vasconcelos
+ */
+@Component(
+	configurationPid = "com.liferay.document.library.internal.configuration.DLSizeLimitConfiguration",
+	property = Constants.SERVICE_PID + "=com.liferay.document.library.internal.configuration.DLSizeLimitConfiguration.scoped",
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/helper/DLSizeLimitConfigurationHelper.java:35]: property = Constants.SERVICE_PID + "=com.liferay.document.library.internal.configuration.DLSizeLimitConfiguration.scoped" is not required here.

File: modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/internal/configuration/admin/service/test/DLSizeLimitManagedServiceFactoryTest.java [hunk 3]
```diff
@@ -190,22 +191,26 @@ public void testGetGroupMimeTypeSizeLimitWithInvalidGroupId()
...
+			_managedServiceFactory, "_dlSizeLimitConfigurationHelper");
 	}
 
 	private long _getGroupMimeTypeSizeLimit(long groupId, String mimeType)
 		throws Exception {
 
 		Method method = ReflectionUtil.getDeclaredMethod(
-			_managedServiceFactory.getClass(), "getGroupMimeTypeSizeLimit",
-			long.class, String.class);
+			_getDLSizeLimitConfigurationHelper().getClass(),
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-test/src/testIntegration/java/com/liferay/document/library/internal/configuration/admin/service/test/DLSizeLimitManagedServiceFactoryTest.java]: avoid chaining

File: modules/apps/document-library/document-library-service/src/test/java/com/liferay/document/library/internal/util/DLValidatorImplTest.java [hunk 4]
```diff
@@ -38,16 +38,16 @@ public void setUp() {
...
-			DLSizeLimitManagedServiceFactory.class);
-
-		dlValidatorImpl.setDLSizeLimitManagedServiceFactory(
-			_dlSizeLimitManagedServiceFactory);
-
 		_groupLocalService = Mockito.mock(GroupLocalService.class);
 
 		dlValidatorImpl.setGroupLocalService(_groupLocalService);
 
+		_dlSizeLimitConfigurationHelper = Mockito.mock(
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/test/java/com/liferay/document/library/internal/util/DLValidatorImplTest.java:45]: create the mock when creating the field.

File: modules/apps/document-library/document-library-service/src/test/java/com/liferay/document/library/internal/util/DLValidatorImplTest.java [hunk 5]
```diff
@@ -38,16 +38,16 @@ public void setUp() {
...
+			DLSizeLimitConfigurationHelper.class);
+
+		dlValidatorImpl.setDLSizeLimitConfigurationHelper(
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/test/java/com/liferay/document/library/internal/util/DLValidatorImplTest.java:48]: sort method call

**Review by rafaprax — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java [hunk 6]
```diff
@@ -165,130 +63,61 @@ public void updated(String pid, Dictionary<String, ?> dictionary)
...
-	}
-
 	private void _unmapPid(String pid) {
 		if (_companyIds.containsKey(pid)) {
 			long companyId = _companyIds.remove(pid);
 
-			_companyConfigurationBeans.remove(companyId);
-			_companyMimeTypeSizeLimitsMap.remove(companyId);
+			_dlSizeLimitConfigurationHelper.removeCompanyConfigurationBeans(
+				companyId);
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java:71]: rename to `removeCompanyConfigurationBean`

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java [hunk 7]
```diff
@@ -165,130 +63,61 @@ public void updated(String pid, Dictionary<String, ?> dictionary)
...
+			_dlSizeLimitConfigurationHelper.removeGroupMimeTypeSizeLimitsMap(
+				companyId);
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java:73]: call the `removeCompanyMimeTypeSizeLimitsMap` instead

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java [hunk 8]
```diff
@@ -165,130 +63,61 @@ public void updated(String pid, Dictionary<String, ?> dictionary)
...
-			});
-	}
-
 	private void _unmapPid(String pid) {
 		if (_companyIds.containsKey(pid)) {
 			long companyId = _companyIds.remove(pid);
 
-			_companyConfigurationBeans.remove(companyId);
-			_companyMimeTypeSizeLimitsMap.remove(companyId);
+			_dlSizeLimitConfigurationHelper.removeCompanyConfigurationBeans(
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/admin/service/DLSizeLimitManagedServiceFactory.java:70]: Hey @RenanUmV. Please move the entire `_unmapPid` to the helper class. Thx

File: modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/persistence/listener/DLSizeLimitConfigurationModelListener.java [hunk 9]
```diff
@@ -63,13 +63,13 @@ public void onBeforeSave(String pid, Dictionary<String, Object> properties)
 		}
 	}
 
-	protected void setDLSizeLimitManagedServiceFactory(
-		DLSizeLimitManagedServiceFactory dlSizeLimitManagedServiceFactory) {
+	protected void setDLSizeLimitConfigurationHelper(
```
Comment (rafaprax) [inline modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/persistence/listener/DLSizeLimitConfigurationModelListener.java:66]: Please remove this method and use `ReflectionTestUtil` to inject the field value into the test classes.

**Review by rafaprax — APPROVED** [review]

**Pull request closed without merging**